tests/unit_tests/analysis/test_adaptive_localization.py

```diff
@@ -102,3 +101,63 @@ def test_that_adaptive_localization_with_cutoff_0_equals_ESupdate(copy_case):
 
     # Check posterior sample without adaptive localization and with cut-off 0 are equal
     assert np.allclose(posterior_sample_loc0, posterior_sample_noloc)
+
+
+@pytest.mark.integration_test
+@pytest.mark.parametrize(
+    "cutoff1, cutoff2",
+    [
+        (0.1, 0.8),
+        (0.3, 0.6),
+        (1e-3, 0.9),
+        (0.5, 0.505),
+        (1e-5, 1e-2),
+    ],
+)
+def test_that_posterior_generalized_variance_increases_in_cutoff(
+    copy_poly_case, cutoff1, cutoff2
+):
+    set_adaptive_localization_cutoff1 = dedent(
+        f"""
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION True
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION_CORRELATION_THRESHOLD {cutoff1}
+        """
+    )
+    set_adaptive_localization_cutoff2 = dedent(
+        f"""
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION True
+        ANALYSIS_SET_VAR STD_ENKF LOCALIZATION_CORRELATION_THRESHOLD {cutoff2}
+        """
+    )
+
+    with open("poly.ert", "r+", encoding="utf-8") as f:
+        lines = f.readlines()
+        lines.insert(2, random_seed_line)
+        lines.insert(9, set_adaptive_localization_cutoff1)
+
+    with open("poly_localization_cutoff1.ert", "w", encoding="utf-8") as f:
+        f.writelines(lines)
+
+    lines.remove(set_adaptive_localization_cutoff1)
+    lines.insert(9, set_adaptive_localization_cutoff2)
+    with open("poly_localization_cutoff2.ert", "w", encoding="utf-8") as f:
+        f.writelines(lines)
+
+    prior_sample_cutoff1, posterior_sample_cutoff1 = run_cli_ES_with_case(
+        "poly_localization_cutoff1.ert"
+    )
+    prior_sample_cutoff2, posterior_sample_cutoff2 = run_cli_ES_with_case(
+        "poly_localization_cutoff2.ert"
+    )
+
+    prior_cov = pd.DataFrame(prior_sample_cutoff1.values).cov().to_numpy()
```

Comment:
I would personally just use [numpy.cov](https://numpy.org/doc/stable/reference/generated/numpy.cov.html). No reason to introduce a DataFrame and leave numpy-land if all you need is the covariance.